--- s3boto.py	2012-08-17 17:21:51.000000000 -0700
+++ s3boto.py~	2012-08-17 16:12:34.000000000 -0700
@@ -333,14 +333,14 @@
         # Parse the last_modified string to a local datetime object.
         return _parse_datestring(entry.last_modified)
 
-    def url(self, name):
+    def url(self, name, response_headers=None):
         name = self._normalize_name(self._clean_name(name))
         if self.custom_domain:
             return "%s://%s/%s" % ('https' if self.secure_urls else 'http',
                                    self.custom_domain, name)
         return self.connection.generate_url(self.querystring_expire,
             method='GET', bucket=self.bucket.name, key=self._encode_name(name),
-            query_auth=self.querystring_auth, force_http=not self.secure_urls)
+            query_auth=self.querystring_auth, response_headers=response_headers, force_http=not self.secure_urls)
 
     def get_available_name(self, name):
         """ Overwrite existing file with the same name. """
